pkgs/development/python-modules/openapi-core/default.nix

```diff
@@ -1,56 +1,55 @@
 { lib
-, attrs
 , buildPythonPackage
-, dictpath
 , django
 , djangorestframework
 , falcon
 , fetchFromGitHub
+, fetchpatch
 , flask
 , isodate
-, lazy-object-proxy
-, mock
 , more-itertools
 , openapi-schema-validator
 , openapi-spec-validator
 , parse
+, poetry-core
 , pytestCheckHook
 , pythonOlder
 , responses
-, six
 , webob
 , werkzeug
 , python
 }:
 
 buildPythonPackage rec {
   pname = "openapi-core";
-  version = "0.14.2";
-  format = "setuptools";
+  version = "0.15.0";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "p1c2u";
     repo = "openapi-core";
     rev = version;
-    hash = "sha256-+VyNPSq7S1Oz4eGf+jaeRTx0lZ8pUA+G+KZ/5PyK+to=";
+    hash = "sha256-7RjKll+z9wrax9hLS4cKpQLOvmlD+lXbk9CkZi0oSbw=";
   };
 
-  postPatch = ''
-    sed -i "/^addopts/d" setup.cfg
-  '';
+  patches = [
+    # Fix for openapi-spec-validator >=0.5.0
+    (fetchpatch {
+      name = "switch-to-jsonschema-spec.patch";
+      url = "https://github.com/p1c2u/openapi-core/commit/202c5b62b58b488eb9ce04a7791623748f23f19b.patch";
+      hash = "sha256-3vc4QnQEi96xUESo+27qQlzPMvC0FS7qZmXy8AWZt28=";
+    })
+  ];
 
+  nativeBuildInputs = [ poetry-core ];
   propagatedBuildInputs = [
-    attrs
-    dictpath
     isodate
-    lazy-object-proxy
     more-itertools
     openapi-schema-validator
     openapi-spec-validator
     parse
-    six
     werkzeug
   ];
```

Comment:
I think this requires `pathable` (in place of `dictpath`) and, on Python 3.8 and lower, `typing-extensions`.

Reply:
Does it also require `jsonschema-spec` after applying the patch?

Reply:
`python38Packages.openapi-core` cannot be built because `pkgutil-resolve-name` is missing (required dep for Python <3.9).
I added the missing `propagatedBuildInputs`, thank you!